src/smc-webapp/project_files.cjsx

```diff
@@ -1461,10 +1490,69 @@ ProjectFilesSearch = rclass
         selected_file_index : 0
         num_files_displayed : 0
 
+    getInitialState : ->  # Miniterm functionality
+        stdout : undefined
+        state  : 'edit'   # 'edit' --> 'run' --> 'edit'
+        error  : undefined
+
+    # Miniterm functionality
```

Comment:
Yup. No idea what will eventually happen to the actual miniterm component. It thought it was potentially getting axed later. Miniterm and its functionality is used no where else. It's certainly not DRY but I don't see a reason for it to be when the parts are well commented and the added functionality itself is experimental. I didn't see the value in abstracting in this case. 

If we want to mimic this in more places, I'm all for moving it out.